eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -56,6 +56,50 @@ jobs:
 
   # Disable build for cpp - client
   - ${{ if ne(parameters.ServiceDirectory, 'not-specified' )}}:
```

Comment:
orthogonal but depending on not-specified here seems wrong.

Reply:
Seems like we should just make `ServiceDirectory` a required parameter. I'm not seeing any usages of it where it isn't passed in. @danieljurek any historical context here I need to be aware of?

Reply:
I think we deleted the old `cpp - client` pipeline as well, so this whole conditional can probably be removed.

Reply:
`cpp - client` has been retired for a very long time. 

This condition can be safely removed.